agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/GeneExpressionAnnotationIndexer.java

```diff
@@ -61,6 +70,21 @@ protected void startSingleThread(LinkedBlockingDeque<String> queue) {
 				List<GeneExpressionAnnotationDocument> documentsToIndex = new ArrayList<>();
 				for (GeneExpressionAnnotation gea : response.getResults()) {
 					GeneExpressionAnnotationDocument geneExpressionAnnotationDocument = new GeneExpressionAnnotationDocument();
+					if (gea.getDataProvider().getAbbreviation().equals("MGI") || gea.getDataProvider().getAbbreviation().equals("WB")) {
+						ConsolidatedGeneExpressionFmsDTO geneExpressionFmsDTO = new ConsolidatedGeneExpressionFmsDTO();
+						geneExpressionFmsDTO.setGeneId(gea.getExpressionAnnotationSubject().getPrimaryExternalId());
+						geneExpressionFmsDTO.setAssay(gea.getExpressionAssayUsed().getCurie());
+						String experimentId = geneExpressionAnnotationUniqueIdHelper.generateExperimentId(geneExpressionFmsDTO, gea.getEvidenceItem().getCurie());
+						SearchResponse<GeneExpressionExperiment> experimentResponse = geneExpressionExperimentService.getGeneExpressionExperiment(0, 10, experimentId);
```

Comment:
In the last round of indexers we made the choice to more the document creation to the curation server. I think this code would become much simpler if it was moved there.

Reply:
Plus pulling these 1 at a time is going to be incredibility slow.